src/plugin.js

```diff
@@ -320,37 +322,98 @@ const plugin = function(options) {
+    return items.map(o => {
+      const mergeOptions = videojs.mergeOptions(settings, o);
+      const attachToControlBar = typeof mergeOptions.attachToControlBar === 'string' || mergeOptions.attachToControlBar === true;
 
-    if (!this.controls() || !this.controlBar) {
-      return this.addChild('overlay', mergeOptions);
-    }
+      if (!this.controls() || !this.controlBar) {
+        return this.addChild('overlay', mergeOptions);
+      }
 
-    if (attachToControlBar && mergeOptions.align.indexOf('bottom') !== -1) {
-      let referenceChild = this.controlBar.children()[0];
+      if (attachToControlBar && mergeOptions.align.indexOf('bottom') !== -1) {
+        let referenceChild = this.controlBar.children()[0];
 
-      if (this.controlBar.getChild(mergeOptions.attachToControlBar) !== undefined) {
-        referenceChild = this.controlBar.getChild(mergeOptions.attachToControlBar);
-      }
+        if (this.controlBar.getChild(mergeOptions.attachToControlBar) !== undefined) {
+          referenceChild = this.controlBar.getChild(mergeOptions.attachToControlBar);
+        }
 
-      if (referenceChild) {
-        const referenceChildIndex = this.controlBar.children().indexOf(referenceChild);
-        const controlBarChild = this.controlBar.addChild('overlay', mergeOptions, referenceChildIndex);
+        if (referenceChild) {
+          const referenceChildIndex = this.controlBar.children().indexOf(referenceChild);
+          const controlBarChild = this.controlBar.addChild('overlay', mergeOptions, referenceChildIndex);
 
-        return controlBarChild;
+          return controlBarChild;
+        }
       }
+
+      const playerChild = this.addChild('overlay', mergeOptions);
+
+      this.el().insertBefore(
+        playerChild.el(),
+        this.controlBar.el()
+      );
+
+      return playerChild;
+    });
+  };
+
+  this.overlays_ = mapOverlays(overlays);
+
+  /**
+   * Adds one or more items to the existing list of overlays.
+   *
+   * @param {Object|Array} item
+   *        An item (or an array of items) to be added as overlay/s
+   *
+   * @return {Array[Overlay]}
+   *         The array of overlay objects that were added
+   */
+  function add(item) {
+    if (!Array.isArray(item)) {
+      item = [item];
     }
 
-    const playerChild = this.addChild('overlay', mergeOptions);
+    const addedOverlays = mapOverlays(item);
+
+    self.overlays_ = self.overlays_.concat(addedOverlays);
+
+    return addedOverlays;
+  }
+
+  /**
+   *
+   * @param {Overlay} item
+   *        An item to be removed from the array of overlays
+   *
+   * @throws {Error}
+   *        Item to remove must be present in the array of overlays
+   *
+   */
+  function remove(item) {
+    const index = self.overlays_.indexOf(item);
+
+    if (index !== -1) {
+      item.el().parentNode.removeChild(item.el());
+      self.overlays_.splice(index, 1);
+    } else {
+      throw new Error('overlay is invalid and cannot be removed');
```

Comment:
I'm not sure throwing is the right thing here. It may be better to use `player.log.warn` instead.

Reply:
Changed the error throwing to be a log.warn 🥳